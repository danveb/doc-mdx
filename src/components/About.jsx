/* eslint-disable react/no-unescaped-entities */
import { textSections } from "../constants";
import "../styles/About.css";

export default function About() {
  return (
    <div className="about">
      <div className="about__wrapper text__container">
        {textSections.map((section) => (
          <div key={section.id}>
            <div className="about__main">
              {section.head && <p className="title">{section.head}</p>}
              {section.subhead && <p className="subtitle">{section.subhead}</p>}
            </div>
            <div className="about__sub">
              {section.title && <p className="title">{section.title}</p>}
              {section.text && <p className="text">{section.text}</p>}
            </div>
          </div>
        ))}
      </div>
    </div>
  )
}