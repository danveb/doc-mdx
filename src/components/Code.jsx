import PropTypes from "prop-types";
import CopyToClipboard from "react-copy-to-clipboard";
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
import { vscDarkPlus } from "react-syntax-highlighter/dist/esm/styles/prism";
import { CopyIcon, PasteIcon } from "./assets/icons";
import "../styles/Code.css";
import { useEffect, useState } from "react";

export default function Code({ children, language }) {
  // useState 
  const [copied, setCopied] = useState(false);

  // useEffect
  useEffect(() => {
    const timer = setTimeout(() => {
      setCopied(false);
    }, 1500);
    return () => clearTimeout(timer);
  }, [copied]);

  return (
    <div className="code">
      <CopyToClipboard text={children} onCopy={() => setCopied(true)}>
        <button className="code__icon copy__icon">
          {copied ? <PasteIcon className="paste__icon" /> : <CopyIcon />}
        </button>
      </CopyToClipboard>
      <SyntaxHighlighter
        language={language}
        style={vscDarkPlus}
        wrapLongLines={true}
      >
        {children}
      </SyntaxHighlighter>
    </div >
  )
}

// PropTypes object to be defined to fix missing props validation
Code.propTypes = {
  children: PropTypes.node.isRequired,
  language: PropTypes.string.isRequired,
}