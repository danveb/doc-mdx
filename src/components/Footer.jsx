import { Link } from "react-router-dom";
import "../styles/Footer.css";
import gitHub from "./assets/github-mark-white.png";

export default function Footer() {
  return (
    <div className="footer">
      <div className="footer__wrapper container">
        <p className="mini__text">Â© All rights reserved |  danveb</p>
        <Link to="https://www.github.com/danveb" target="_blank" rel="noreferrer noopener">
          <img src={gitHub} alt="github logo" />
        </Link>
      </div>
    </div>
  )
}