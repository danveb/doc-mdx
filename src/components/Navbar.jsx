import { LogoIcon, MoonIcon } from "./assets/icons";
import "../styles/Navbar.css";

export default function Navbar() {
  return (
    <header className="navbar">
      <div className="nav__wrapper container">
        <div className="nav__logo">
          <LogoIcon />
          <p>doc-mdx</p>
          <button className="nav__icon"><MoonIcon /></button>
        </div>
      </div>
    </header>
  )
}