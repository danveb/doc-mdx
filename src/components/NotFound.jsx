import "../styles/NotFound.css";

export default function NotFound() {
  return (
    <div className="not__found container">
      <p className="text">404 | Sorry, page not found.</p>
    </div>
  )
}