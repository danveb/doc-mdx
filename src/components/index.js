export { default as About } from "./About";
export { default as Code } from "./Code";
export { default as Navbar } from "./Navbar";
export { default as NotFound } from "./NotFound";
export { default as Post } from "./Post"; 