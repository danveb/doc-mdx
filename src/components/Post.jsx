import Markdown from "markdown-to-jsx";
import { useEffect, useState } from "react";
import { Code } from "../components";

export default function Post() {
  // useState
  const [posts, setPosts] = useState("");

  useEffect(() => {
    fetchMarkdownFile();
  }, []);

  const fetchMarkdownFile = () => {
    import("../markdown/article.md")
      .then(res => {
        fetch(res.default)
          .then(res => res.text())
          .then(res => setPosts(res))
          .catch(err => console.log(err));
      })
      .catch(err => console.log(err));
  };

  return (
    <article className="article">
      <div className="article__wrapper container">
        <div className="post__wrapper">
          <Markdown options={{
            overrides: {
              Code: {
                component: Code,
              },
            },
          }}>{posts}</Markdown>
        </div>
      </div>
    </article>
  )
}