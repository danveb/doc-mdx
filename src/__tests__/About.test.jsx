import { describe, test, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { About } from "../components";

describe("About component", () => {
  test("renders without crashing", () => {
    const { container } = render(<About />);
    expect(container).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("renders correct paragraph on component", () => {
    render(
      <About />
    )
    const text = screen.getByText("This is doc-mdx, a simple application that allows users create concise technical documentation directly from a markdown document. In this article, I will go over the process of installation and usage.");
    expect(text).toBeInTheDocument();
  });
});