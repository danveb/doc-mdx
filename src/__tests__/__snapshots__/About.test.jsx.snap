// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="about"
  >
    <div
      class="about__wrapper text__container"
    >
      <div>
        <div
          class="about__main"
        >
          <p
            class="title"
          >
            Danny Bae
          </p>
          <p
            class="subtitle"
          >
            Frontend Developer
          </p>
        </div>
        <div
          class="about__sub"
        />
      </div>
      <div>
        <div
          class="about__main"
        />
        <div
          class="about__sub"
        >
          <p
            class="title"
          >
            Hello World
          </p>
          <p
            class="text"
          >
            This is doc-mdx, a simple application that allows users create concise technical documentation directly from a markdown document. In this article, I will go over the process of installation and usage.
          </p>
        </div>
      </div>
      <div>
        <div
          class="about__main"
        />
        <div
          class="about__sub"
        >
          <p
            class="title"
          >
            Markdown
          </p>
          <p
            class="text"
          >
            Markdown is a lightweight markup language widely used for formatting plain text documents. It is used among developers who need a simple and highly readable way to craft documentation related to a code base. Users can use markdown syntax or even HTML syntax to build documents.
          </p>
        </div>
      </div>
      <div>
        <div
          class="about__main"
        />
        <div
          class="about__sub"
        >
          <p
            class="title"
          >
            doc-mdx
          </p>
          <p
            class="text"
          >
            This project incorporates markdown into React.js.
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
